orbit/pkg/useraction/mdm_migration_darwin.go

```diff
@@ -326,7 +338,21 @@ func (m *swiftDialogMDMMigrator) waitForUnenrollment() error {
 }
 
 func (m *swiftDialogMDMMigrator) renderMigration() error {
-	message, flags, err := m.getMessageAndFlags()
+	log.Debug().Msg("checking manual enrollment status")
+	manualProfileCheck, err := profiles.IsManuallyEnrolledInMDM()
+	if err != nil {
+		return err
+	}
+
+	// Check if we're in a manual migration.
+	migrationType, err := m.mrw.GetMigrationType()
```

Comment:
I like the idea of using a flag to signal that a migration is in process instead of storing the whole config if you think that's too much.

I don't like that we're hardcoding the migration type here because it seems hard to get right. I'm thinking in cases where the computer enrolled manually, but it's actually assigned in ABM to Fleet, so you'd want an ADE enrollment there.

A safer check is perhaps:

1. The migration file exists
2. The host can enroll using ADE, check here:

https://github.com/fleetdm/fleet/blob/3686149391a9d3cc35e9ea324351e412e906846f/orbit/pkg/update/notifications.go#L90-L94

what do you think?

Reply:
If the host is enrolled manually, but assigned to Fleet via ABM, would that change the result [here](https://github.com/fleetdm/fleet/blob/6fce9d5224aca4c67f2f716369cc9a75bfb41894/orbit/pkg/profiles/profiles_darwin.go#L130)? So like the first like of the output would be `Enrolled via DEP: Yes`, which would tell us that we're not doing a manual migration.

Regardless, I definitely see your point about the code being too closely tied to the manual case. I'm thinking about how to fix that.

Reply:
this is what I'm thinking: there are two ways to think about this problem:

1. See how the host was previously enrolled -> try to replicate that after the migration (current approach)
2. After migration, figure out how should we enroll the host

I'm suggesting `2` because it seems safer (no need to carry state before/after migration) and addresses edge cases, but I don't think it's a blocker.